src/NuGet.Core/NuGet.Common/InProcLock.cs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Common
+{
+    internal class InProcLock : IDisposable
+    {
+        private Dictionary<string, LockState> _locks;
+        private SemaphoreSlim _dictionaryLock;
+
+        internal InProcLock()
+        {
+            _locks = new Dictionary<string, LockState>();
+            _dictionaryLock = new SemaphoreSlim(1);
```

Comment:
Add a named parameter to explain that mysterious integer literal.

Reply:
```csharp
int mysteryValue = 1;
_dictionaryLock = new SemaphoreSlim(initialCount: mysteryValue);
```

Reply:
suggestion: 
```cs
var mysteryValue = 1; 
```

Reply:
Consider adding obfuscation:

```suggestion
var mysteryValue = 5 & 3;
```